File: entities/job.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type JobStatus string
type JobType string

const (
	JobStatusQueued     JobStatus = "QUEUED"
	JobStatusProcessing JobStatus = "PROCESSING"
	JobStatusRunning    JobStatus = "PROCESSING"
	JobStatusCompleted  JobStatus = "COMPLETED"
	JobStatusSuccess    JobStatus = "COMPLETED"
	JobStatusFailed     JobStatus = "FAILED"

	JobTypeFullPipeline JobType = "FULL_PIPELINE"
	JobTypeOCROnly      JobType = "OCR_ONLY"
	JobTypeOCR          JobType = "OCR_ONLY"
	JobTypeCouncilOnly  JobType = "COUNCIL_ONLY"
)

type VerificationJob struct {
	JobID        uuid.UUID  `gorm:"primaryKey" json:"job_id"`
	DoctorID     uuid.UUID  `gorm:"not null;index" json:"doctor_id"`
	JobType      JobType    `gorm:"size:50;default:'FULL_PIPELINE';not null" json:"job_type"`
	Status       JobStatus  `gorm:"size:50;default:'QUEUED';not null;index" json:"status"`
	Priority     int        `gorm:"default:1;not null;index" json:"priority"`
	RetryCount   int        `gorm:"default:0;not null" json:"retry_count"`
	MaxRetries   int        `gorm:"default:3;not null" json:"max_retries"`
	ErrorMessage *string    `gorm:"type:text" json:"error_message,omitempty"`
	StartedAt    *time.Time `json:"started_at,omitempty"`
	CompletedAt  *time.Time `json:"completed_at,omitempty"`
	CreatedAt    time.Time  `gorm:"autoCreateTime;index" json:"created_at"`
	UpdatedAt    time.Time  `gorm:"autoUpdateTime" json:"updated_at"`
}

func (j *VerificationJob) BeforeCreate(tx *gorm.DB) (err error) {
	if j.JobID == uuid.Nil {
		j.JobID = uuid.New()
	}
	return nil
}

File: backend_api/entities/job.py
import enum
import uuid
from sqlalchemy import Column, String, Integer, DateTime, Text, Enum, ForeignKey
from sqlalchemy.sql import func
from database import Base

class JobStatus(str, enum.Enum):
    QUEUED = "QUEUED"
    PROCESSING = "PROCESSING"
    RUNNING = "PROCESSING"
    COMPLETED = "COMPLETED"
    SUCCESS = "COMPLETED"
    FAILED = "FAILED"

class JobType(str, enum.Enum):
    FULL_PIPELINE = "FULL_PIPELINE"
    OCR_ONLY = "OCR_ONLY"
    OCR = "OCR_ONLY"
    COUNCIL_ONLY = "COUNCIL_ONLY"

class VerificationJob(Base):
    __tablename__ = "verification_jobs"

    job_id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    job_type = Column(Enum(JobType, name="job_type_enum", native_enum=False), default=JobType.FULL_PIPELINE, nullable=False)
    status = Column(Enum(JobStatus, name="job_status_enum", native_enum=False), default=JobStatus.QUEUED, nullable=False, index=True)
    priority = Column(Integer, default=1, nullable=False, index=True)
    retry_count = Column(Integer, default=0, nullable=False)
    max_retries = Column(Integer, default=3, nullable=False)
    error_message = Column(Text, nullable=True)
    started_at = Column(DateTime(timezone=True), nullable=True)
    completed_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), index=True)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
